ngraph/python/src/pyngraph/frontend_manager.cpp

```diff
@@ -99,3 +101,15 @@ void regclass_pyngraph_FEC(py::module m)
            const ngraph::frontend::FrontEndCapabilities& b) { return a == b; },
         py::is_operator());
 }
+
+void regclass_pyngraph_CheckFailureFrontEnd(py::module m) {
+    static py::exception<ngraph::frontend::CheckFailureFrontEnd> exc(m, "CheckFailureFrontEnd");
+    py::register_exception_translator([](std::exception_ptr p) {
+        try {
+            if (p) std::rethrow_exception(p);
+        } catch (const ngraph::frontend::CheckFailureFrontEnd &e) {
+            exc(e.what());
```

Comment:
Not clear for me how Model optimizer can handle such exceptions smoothly?
E.g. when 'convert' throws exception, how MO can understand that this is due to there are some dynamic shapes in a graph? Current MO behavior produces very informative messages with links to FAQ (like https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html#40).
So, to keep this behavior for new frontends we should have a kind of 'error code' in the exception, not just text string

Reply:
Why error code is better than string?
The message will be looks like: "An issue occurs in "some_file.cpp" line N. Dynamic shapes check failed. 
Node, shapes can be printed. 
Not sure, but I think we can provide informative messages using c++, not python.

Reply:
I mean for example to have some 'enum' with error codes exposed to python. And MO can catch and check error code of exception. like (if ex.code == FrontEndErrorCode.DYNAMIC_SHAPE)
Alternatively we can probably have different 'exception classes', like class DynamicShapeNotSupported: public std::runtime_error {...}

Reply:
Why do we need this checks like this `(if ex.code == FrontEndErrorCode.DYNAMIC_SHAPE)` in MO?  
All context information is already in the message from c ++

Reply:
I agree with @itikhono c++ should return verbose message, because not only MO should tell user what happened, direct input through reader API should also tell user whats wrong.

Reply:
done